### PR TITLE
samples: add Wio Terminal overlay to blinky pwm

### LINKED ISSUE #66547: wio-terminal board does not support PWM?
Hey,

I am currently looking into the wio-terminal, which is listed as [having support for pwm](https://github.com/zephyrproject-rtos/zephyr/blob/29b19bcf8bc8f97069be00a9012c4cf1b7dbfb8b/boards/arm/wio_terminal/doc/index.rst?plain=1#L88-L90). But the devicetree is built as follows:

1. dts: [boards/arm/wio_terminal/wio_terminal.dts](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/wio_terminal/wio_terminal.dts)
2. via `#include <atmel/samd5xx19.dtsi>` this includes: [dts/arm/atmel/samd5xx19.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samd5xx19.dtsi)
3. via `#include <atmel/samd5x.dtsi>` this includes: [dts/arm/atmel/samd5x.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samd5x.dtsi)

and in the file [samd5x.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/atmel/samd5x.dtsi) all nodes `tcc*` are `compatible = "atmel,sam0-tcc";` instead of `"atmel,sam0-tcc-pwm"` which, as far as I understand means that they do not support PWM?

### PR DESCRIPTION
**Pull request by joelguittet**
Add overlay to the blinky pwm sample to run the application on Wio Terminal. The USER LED on PA15 is used to setup the PWM LED.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/66547

### CODE REVIEW

**Comment by kartben** [comment]
Thanks! This could use a review from folks knowledgeable about Microchip SAMD drivers (cc @nandojve) to make sure this looks correct.

I just tested it and while it works, one thing that I am not sure is if the timing is as expected: should the LED initially blink at 1Hz as in 1sec-on/1sec-off, or should it be (like it seems to be the case right now) .5sec on/.5 sec off, i.e. 1Hz for a "full" blink.

**Comment by joelguittet** [comment]
Good point, to be honest I m not 100% sure because I tried to understand the syntax of the PWM entries in DTS, but I don't clearly understand the doc on that.

By the way writing PWM 500ms I expect the frequency of the output to be 2hz. I.e. the period is 500ms.

I haven't checked that, I may focused on the setup with this overlay and then I focused on what this sample does (changing the frequency periodically)

**Comment by kartben** [comment]
> Good point, to be honest I m not 100% sure because I tried to understand the syntax of the PWM entries in DTS, but I don't clearly understand the doc on that.
 
Same here, hence why hopefully others can chime in :) Maybe @anangl and @henrikbrixandersen as PWM maintainers/collaborators can help shed some light too

> [...] and then I focused on what this sample does (changing the frequency periodically)

Ya, at least I can confirm that it does that correctly :)

**Review by nandojve — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
> Same here, hence why hopefully others can chime in :) Maybe @anangl and @henrikbrixandersen as PWM maintainers/collaborators can help shed some light too

If you are using `samples/basic/blinky_pwm/` you need to check the console output, which should contain details about the "calibrated" period, and compare it against the actual period observed.

I do not have access to the hardware, but the change looks good to me.

**Comment by joelguittet** [comment]
@henrikbrixandersen I haven't used a scope, but seems 500ms make the led blinking twice per second so 2hz visually. So looks good to me.

**Comment by kartben** [comment]
Where I'm confused is that other overlays are setting a much shorter period, ex. 20 ms, so I am a vmbit lost as to what the period specified in the device tree is really meant for.

**Comment by henrikbrixandersen** [comment]
> Where I'm confused is that other overlays are setting a much shorter period, ex. 20 ms, so I am a vmbit lost as to what the period specified in the device tree is really meant for.

The period in the devicetree can be used from the application. Typically, we set a period of 20msec for PWM controlled LEDs.

**Comment by joelguittet** [comment]
Yes to drive backlight on LCD for example.

**Comment by joelguittet** [comment]
Seems approval are given here, who is able to merge ?

**Comment by dleach02** [comment]
@kartben is the assignee so need him to wrap up the approvals

**Comment by kartben** [comment]
> > Where I'm confused is that other overlays are setting a much shorter period, ex. 20 ms, so I am a vmbit lost as to what the period specified in the device tree is really meant for.
> 
> The period in the devicetree can be used from the application. Typically, we set a period of 20msec for PWM controlled LEDs.

Ya but I still don't understand why in blinky_pwm some boards would define the PWM with `PWM_MSEC(20)`, some with 1000, and now one that does 500. Is the number provided in the devicetree completeley irrelevant in the context of what the sample then actually do re: autocalibration etc or something?

**Review by kartben — DISMISSED** [review]

**Review by kartben — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**
